utils/filteroutput.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+#
+# A thin Python wrapper around addr2line, can monitor esp-open-rtos
+# output and uses gdb to convert any suitable looking hex numbers
+# found in the output into function and line numbers.
+#
+# Works with a serial port if the --port option is supplied.
+# Otherwise waits for input on stdin.
+#
+import serial
+import argparse
+import re
+import os
+import os.path
+import subprocess
+import termios
+import sys
+import time
+
+# Try looking up anything in the executable address space
+RE_EXECADDR = r"(0x)?40([0-9]|[a-z]){6}"
+
+def find_elf_file():
+    out_files = []
+    for top,_,files in os.walk('.', followlinks=False):
+        for f in files:
+            if f.endswith(".out"):
+                out_files.append(os.path.join(top,f))
+    if len(out_files) == 1:
+        return out_files[0]
+    elif len(out_files) > 1:
+        print("Found multiple .out files: %s. Please specify one with the --elf option." % out_files)
+    else:
+        print("No .out file found under current directory. Please specify one with the --elf option.")
+    sys.exit(1)
+
+def main():
+    parser = argparse.ArgumentParser(description='esp-open-rtos output filter tool', prog='filteroutput')
+    parser.add_argument(
+        '--elf', '-e',
+        help="ELF file (*.out file) to load symbols from (if not supplied, will search for one)"),
+    parser.add_argument(
+        '--port', '-p',
+        help='Serial port to monitor (will monitor stdin if None)',
+        default=None)
+    parser.add_argument(
+        '--baud', '-b',
+        help='Baud rate for serial port',
+        type=int,
+        default=74880)
+    parser.add_argument(
+        '--reset-on-connect', '-r',
+        help='Reset ESP8266 (via DTR) on serial connect',
```

Comment:
It might be good to add a note that reset-on-connect will always happen on Linux anyway, because of the way the serial drivers are written (this option may still be useful on non-Linux platforms, though)..